src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -195,7 +195,9 @@ public object GetProperyValue(object obj, string propertyName)
                 var completed = await CompleteValue(context, context.Schema.FindType(nonNullType.Type), fields, result);
                 if (completed == null)
                 {
-                    throw new ExecutionError("Cannot return null for non-null type. Field: {0}".ToFormat(nonNullType.Name));
```

Comment:
I agree that the `Name` issue needs to be fixed.  I think this isn't going to help though, since you are essentially picking a random field?  The real field that it is looking for is only known from the caller (which could be several calls deep with lists/non-nulls).  Is this not the case?  Need tests to prove this behavior.  ;]